app/models/ability.rb

```diff
@@ -76,9 +76,11 @@ def signed_in(user)
     can :manage, Event do |event|
       event.users.include?(user)
     end
-    # can create an event until the last day of a conference
-    can :create, Event, program_id: Conference.where('end_date >= ?', Date.today).map { |conference| conference.program.id}.compact
 
+    # can create an event if program has call for papers defined
+    can [:new, :create], Event do |event|
+      event.program.cfp_open? && event.new_record?
```

Comment:
Hmm what I understood is that a user can create an Event, where the user is included in event.users, even if the cfp_open? is false.

From  https://github.com/CanCanCommunity/cancancan/wiki/Ability-Precedence

> Adding can rules do not override prior rules, but instead are logically or'ed.

 (And the example below that)
Shouldn't we add negative tests to ensure that this does not happen?